int32 cmd_left
int32 cmd_right

